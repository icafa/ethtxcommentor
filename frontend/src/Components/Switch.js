import React from 'react'
import { theme } from '@aragon/ui'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { GU } from '../utils/utils'

const StyledSwitch = styled.div`
  height: 20px;
  .switch {
    margin: 0 ${GU / 2}px 0 ${GU / 2}px;
    position: relative;
    display: inline-block;
    width: 40px;
    height: 20px;
    background-color: rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    transition: all 0.3s;
  }

  .switch::after {
    content: '';
    position: absolute;
    width: 18px;
    height: 18px;
    border-radius: 18px;
    background-color: white;
    top: 1px;
    left: 1px;
    transition: all 0.3s;
  }

  input[type='checkbox']:checked + .switch::after {
    transform: translateX(20px);
  }

  input[type='checkbox']:checked + .switch {
    background-color: ${theme.accent};
  }

  .offscreen {
    position: absolute;
    left: -9999px;
  }
`

const Switch = ({ onChange, on }) => (
  <StyledSwitch>
    <input
      type="checkbox"
      id="toggle"
      name="toggle"
      className="offscreen"
      onChange={onChange}
      checked={on}
    />{' '}
    <label htmlFor="toggle" className="switch" />
  </StyledSwitch>
)

Switch.propTypes = {
  onChange: PropTypes.func.isRequired,
  on: PropTypes.bool.isRequired,
}

export default Switch
