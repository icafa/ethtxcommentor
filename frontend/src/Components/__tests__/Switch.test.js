import React from 'react'
import { render } from '@testing-library/react'
import Switch from '../Switch'

describe('Switch component', () => {
  it('renders correctly with on state', () => {
    const onChange = jest.fn()
    const on = true
    const { asFragment } = render(<Switch onChange={onChange} on={on} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders correctly with off state', () => {
    const onChange = jest.fn()
    const on = false
    const { asFragment } = render(<Switch onChange={onChange} on={on} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
