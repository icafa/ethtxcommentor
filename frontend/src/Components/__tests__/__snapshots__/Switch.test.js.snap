// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component renders correctly with off state 1`] = `
<DocumentFragment>
  <div
    class="sc-cHGsZl bFXplU"
  >
    <input
      class="offscreen"
      id="toggle"
      name="toggle"
      type="checkbox"
    />
     
    <label
      class="switch"
      for="toggle"
    />
  </div>
</DocumentFragment>
`;

exports[`Switch component renders correctly with on state 1`] = `
<DocumentFragment>
  <div
    class="sc-cHGsZl bFXplU"
  >
    <input
      checked=""
      class="offscreen"
      id="toggle"
      name="toggle"
      type="checkbox"
    />
     
    <label
      class="switch"
      for="toggle"
    />
  </div>
</DocumentFragment>
`;
